### ISSUE TITLE
`scarb lint` detects more than internal tests

### ISSUE THREAD

**Issue by MKowalski8** [software-mansion/cairo-lint#284]
### Problem

If the following line hadn't been applied:
https://github.com/software-mansion/cairo-lint/blob/19eacd33bb36d9527a31d35cba876bc1044d0750/src/lints/redundant_brackets_in_enum_call.rs#L84

And we run `scarb lint` on the following code:

``` cairo
fn main() {
    format!("abc");
}
```

The linter produces the warning:

```
warn: Plugin diagnostic: redundant parentheses in enum call
 --> PATH/src/lib.cairo[format_macro][write_macro]:3:5
    core::result::Result::<(), core::fmt::Error>::Ok(())
```

However, this issue is not caught by the internal tests.

### Scarb Version

```text
2.11.4
```

### Notes

Tagged: https://github.com/software-mansion/cairo-lint/blob/7a9d455478bfdd05e339a71c2ed70eecd2596717/src/lints/redundant_brackets_in_enum_call.rs#L83
